frontend/src/utils/guard/auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { KeycloakAuthGuard, KeycloakService } from 'keycloak-angular';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard extends KeycloakAuthGuard {
+
+
+  constructor(
+    protected override readonly router: Router,
+    protected readonly keycloak: KeycloakService
+  ) {
+    super(router, keycloak);
+  }
+
+  async isAccessAllowed(
```

Comment:
It will be enough to modify this function.

```
  async isAccessAllowed(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<boolean | UrlTree> {

    if (!this.authenticated) {
      await this.keycloak.login({
        redirectUri: window.location.origin + state.url,
      });
    }

    let roles = route.data['roles'];

    if (!roles || roles.length === 0) {
      return this.authenticated;
    }

    for (let role of roles) {
      if (this.keycloak.isUserInRole(role)) {
        return this.authenticated;
      }
    }
    this.snackBar.open('Access denied', '', {duration: 5000})
    return false;
  }
```